zuul-core/src/main/java/com/netflix/zuul/netty/server/ClientRequestReceiver.java

```diff
@@ -159,6 +158,19 @@ private void channelReadInternal(final ChannelHandlerContext ctx, Object msg) th
                         ZuulStatusCategory.FAILURE_CLIENT_BAD_REQUEST);
                 zuulRequest.getContext().setError(ze);
                 zuulRequest.getContext().setShouldSendErrorResponse(true);
+            } else if (zuulRequest.getHeaders().getAll(HttpHeaderNames.HOST.toString()).size() > 1) {
```

Comment:
This should be `HOST.getName()`

Reply:
HOST is an `AsciiString`, that method doesn't resolve for it.

Reply:
edit: `com.netflix.zuul.message.http.HttpHeaderNames`